### ISSUE TITLE
no-document-write: Error is triggered by call to "write" method on object other than document

### ISSUE THREAD

**Issue by nfriend** [Microsoft/tslint-microsoft-contrib#109]
See the following example:

```typescript
export class MyClass {
    public myMethod = () => {
        let writer = new jsts.io.GeoJSONWriter(),
            geom: jsts.geom.Geometry;

        writer.write(geom);
    }
}
```

When the `no-document-write` rule is `true`, the following error is thrown:

```
test.ts[6, 9]: An error occurred visiting a node.
Walker: NoDocumentWriteWalker
Node: 
            
        writer.write(geom)
TypeError: Cannot read property 'members' of undefined
```

This example requires that the ambient declaration for the `jsts` library be available.: https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/jsts.

**Comment by rovale** [comment]
Hi, in my project (https://github.com/rovale/micro-mqtt) I get a similar error if I enable the `no-document-write` rule:

```
C:\src\micro-mqtt> npm run-script tslint

> micro-mqtt@0.0.6 tslint C:\src\micro-mqtt
> tslint ./src/**/*.ts

src/module/micro-mqtt.ts[107, 9]: An error occurred visiting a node.
Walker: NoDocumentWriteWalker
Node:

        this.networkSocket.write(MqttProtocol.createConnectPacket(this.options))
TypeError: Cannot read property 'flags' of undefined
src/module/micro-mqtt.ts[133, 17]: An error occurred visiting a node.
Walker: NoDocumentWriteWalker
```

I disabled the rule because I will never write in any document, but I just wanted to let you know :-)

**Comment by HamletDRC** [comment]
It seems like the typescript typechecker just doesn't work at all from within tslint. There are so many errors like this that basically everything is an 'any' type.

**Comment by HamletDRC** [comment]
Hi guys, I fixed this issue today by simplifying the rule. You will now receive a violation for: 

    document.write('...'); 
    document.writeln('...'); 
    window.document.write('...'); 
    window.document.writeln('...'); 

But you will *not* receive a violation if document is reassigned to a variable: 

    const doc = document;
    doc.write('...'); // this no longer causes a violation because it often simply caused an error
